src/app/test/test.component.ts

```diff
@@ -246,25 +255,46 @@ export class TestComponent implements OnInit {
   updatePath(action: string) {
     let reportIds: string[] = this.helperService.getSelectedIds(this.reports);
     if (reportIds.length > 0) {
-      let path = (document.querySelector('#moveToInput')! as HTMLInputElement).value;
-      if (!path.endsWith('/')) {
-        path = path + '/';
-      }
-      if (!path.startsWith('/')) {
-        path = '/' + path;
-      }
-      let map: UpdatePathSettings = { path: path, action: action };
+      let path: string = (document.querySelector('#moveToInput')! as HTMLInputElement).value;
+      path = this.addSlashesToPathEnds(path);
+      const map: UpdatePathSettings = { path: path, action: action };
       this.httpService.updatePath(reportIds, this.currentView.storageName, map).subscribe(() => this.loadData(path));
     } else {
       this.toastService.showWarning('No Report Selected!');
     }
   }
 
+  addSlashesToPathEnds(path: string): string {
+    if (path.length > 0 && !path.startsWith('/')) {
+      path = `/ ${path}`;
+    }
+    if (!path.endsWith('/')) {
+      path = `${path}/`;
+    }
+    return path;
+  }
+
   changeFilter(filter: string): void {
+    filter = this.transformFilter(filter);
+    if (filter === this.testFileTreeComponent.rootFolder.name) {
+      filter = '';
+    }
     this.currentFilter = filter;
-    this.reports.forEach((report) => {
+    for (const report of this.reports) {
       report.checked = this.matches(report);
-    });
+    }
+  }
+
+  transformFilter(value: string): string {
+    if (value !== '' && value !== 'Reports') {
+      if (!value.startsWith('/')) {
+        value = `/${value}`;
+      }
+      if (!value.endsWith('/')) {
+        value = `${value}/`;
```

Comment:
Aren't these methods the same? The value assertion could be done in the calling function. Then these two functions can be merged.